### PR TITLE
feat: port to app-platform

### PR DESCRIPTION
**Pull request by mediremi**
Closes https://jira.dhis2.org/browse/DHIS2-10482

# Changes

## Required due to port
- Replaced SCSS with CSS modules
- Renamed `componentWillReceiveProps` to `UNSAFE_componentWillReceiveProps` and `componentWillUpdate` to `UNSAFE_componentWillUpdate`
- Replaced deprecated React v15 PropTypes with `@dhis2/prop-types`
  + Added some missing PropTypes
- Fixed import paths that were broken due to lack of [`webpack resolve`](https://webpack.js.org/configuration/resolve/) plugin in `app-platform`
- Updated `jsoneditor` in order to fix rendering issues with new React and styles
  + Also imported CSS from npm module instead of checked-in CSS file
- Fixed `componentWillReceiveProps` of `JSONEditor` by shallow comparing previous props and current props instead of comparing by reference
  + Incorrect implementation led to broken tree view after updating React
- Replaced [custom spinner](https://github.com/dhis2/datastore-app/blob/db71c1c3df8fa3ff7a05a2f29c59fba964ca9a54/webapp/js/components/utils/Loaders.js) with [`CircularLoader` from `@dhis2/ui`](https://github.com/dhis2/datastore-app/blob/95b814352f353309ca6fa16836fe3650df68f108/src/components/utils/Loaders.js) due to custom spinner using SCSS mixins for styling

## Functional changes
- Removed `NavigationBar` as it was superseded by app platform `HeaderBar`
- Added eslint, prettier, husky, GitHub workflows
- Removed unused/broken components and inaccessible routes (`webapp/js/components/display/history/*.js` and `webapp/js/components/display/statistics/*.js`)
  + Did not work due to missing `Api` method
- Fixed all eslint issues
- Updated `material-ui` from `0.16.4` to `0.19.0` in order to remove `react-tap-event-plugin` dependency
  + Changed `onTouchTap` handlers to `onClick` for all `material-ui` components
- Added i18n support
- Removed unused 'history' actions and reducers (see commit https://github.com/dhis2/datastore-app/commit/4d37fbf8052f1ec7e648844179d3e20a75f85c49)
- Refactored `Api` class ([old](https://github.com/dhis2/datastore-app/blob/db71c1c3df8fa3ff7a05a2f29c59fba964ca9a54/webapp/js/utils/api.js), [new](https://github.com/dhis2/datastore-app/blob/4cb66ef30fab59796eaba9e0d6ec6add3017f66e/src/utils/api.js))
  + Removed unused methods and properties
  + Used async fns to make logic easier to follow
  + Extraced cache-related code to separate class
     - Fix cache implementation by using `Map`
       + `Map` has a `.has(key)` method, which does not suffer from the same problems as `key in object` (i.e. do not have to use `hasOwnProperty`, which [may or may not exist](https://eslint.org/docs/rules/no-prototype-builtins) and triggers a eslint error)
     - [Added tests](https://github.com/dhis2/datastore-app/blob/1a9f5a360d898610e3087800ab5b82a03f490516/src/utils/cache.test.js)
- Replaced `this.fn = this.fn.bind(this)` pattern in React class components to arrow functions

# Further work to do in the future
- Replace `material-ui` with `@dhis2/ui`
- Replace `d2` with `@dhis2/app-runtime` and remove `@dhis2/app-runtime-adapter-d2`
- Setup Transifex
- Replace `react-router` with `react-router-dom`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,16 +0,0 @@
-# [1.1.0](https://github.com/dhis2/datastore-app/compare/v1.0.0...v1.1.0) (2021-02-12)
```
Comment (amcgee) [inline CHANGELOG.md:1]: This shouldn't be deleted, I don't think...

File: LICENSE [hunk 2]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
Comment (amcgee) [inline LICENSE:1]: This should be replaced with our standard BSD-3-Clause license

File: package.json [hunk 3]
```diff
@@ -1,85 +1,37 @@
...
+        "@dhis2/prop-types": "^2.0.3",
+        "d2": "^31.9.0",
+        "jsoneditor": "^9.2.0",
+        "material-ui": "0.19.0",
+        "react-redux": "^7.2.2",
+        "react-router": "3.0.5",
+        "react-router-dom": "^5.2.0",
+        "redux": "^4.0.5",
+        "redux-thunk": "^2.3.0"
+    }
```
Comment (amcgee) [inline package.json:36]: These package.json diffs always make me happy!  🥇

**Comment by amcgee** [comment]
Awesome work here @mediremi !  On first pass this looks really good.  Huge PR though, from the commit messages it looks like there might be some functional changes or fixes mixed in here too... were those all issues caused by the port?  Can you quickly describe any functional changes made here too?  Thanks!!

File: CHANGELOG.md (hunk 1 above)
Comment (mediremi) [reply CHANGELOG.md]: Oooops :sweat_smile:

**Comment by mediremi** [comment]
@amcgee I've updated the PR description with a detailed breakdown of changes. Unfortunately the diff looks much bigger than it should be due to formatting and import changes causing git to show file renames as two separate files instead :disappointed: I should probably use `git mv` next time I port an app to the app platform :thinking:

**Review by Mohammer5 — APPROVED** [review]

**Pull request merged**